### PR TITLE
SoA: Public Getter for Names

### PR DESCRIPTION
**Pull request by ax3l**
## Summary

Expose the names to public member functions.

## Additional background

Needed for Python bindings. Follow-up to #4163 

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Src/Particle/AMReX_StructOfArrays.H [hunk 1]
```diff
@@ -52,6 +53,28 @@ struct StructOfArrays {
...
     [[nodiscard]] const std::array< IntVector,  NInt>& GetIntData  () const { return m_idata; }
 
+    /** Get the names for the real SoA components **/
+    [[nodiscard]] std::vector<std::string> GetRealNames () const
+    {
+        if (m_rdata_names) {
+            return *m_rdata_names;
+        }
+        else {
+            throw std::runtime_error("SoA: no Real names defined!");
```
Comment (ax3l) [inline Src/Particle/AMReX_StructOfArrays.H]: We could also return:
```suggestion
            return std::vector<std::string>();
```

File: Src/Particle/AMReX_StructOfArrays.H [hunk 2]
```diff
@@ -52,6 +53,28 @@ struct StructOfArrays {
...
+    }
+
+    /** Get the names for the int SoA components **/
+    [[nodiscard]] std::vector<std::string> GetIntNames () const
+    {
+        if (m_idata_names) {
+            return *m_idata_names;
+        }
+        else {
+            throw std::runtime_error("SoA: no int names defined!");
```
Comment (ax3l) [inline Src/Particle/AMReX_StructOfArrays.H]: We could also return:
```suggestion
            return std::vector<std::string>();
```

**Comment by ax3l** [comment]
Let me know what you prefer.

In fact in Python bindings I prefer the alternative of returning an empty list over having to handle an exception.

File: Src/Particle/AMReX_StructOfArrays.H (hunk 1 above)
Comment (WeiqunZhang) [reply Src/Particle/AMReX_StructOfArrays.H]: I think return is better.

File: Src/Particle/AMReX_StructOfArrays.H [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 #include <algorithm>
 #include <array>
+#include <stdexcept>
```
Comment (ax3l) [inline Src/Particle/AMReX_StructOfArrays.H]:
```suggestion
```

**Review by atmyers — APPROVED** [review]

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**